engine/src/main/java/org/terasology/rendering/shader/ShaderParametersBase.java

```diff
@@ -43,27 +45,29 @@ public void applyParameters(Material program) {
         program.setFloat("viewingDistance", CoreRegistry.get(Config.class).getRendering().getViewDistance().getChunkDistance().x * 8.0f);
 
         WorldRenderer worldRenderer = CoreRegistry.get(WorldRenderer.class);
-        WorldProvider worldProvider = CoreRegistry.get(WorldProvider.class);
+        BackdropProvider backdropProvider = CoreRegistry.get(BackdropProvider.class);
 
-        if (worldRenderer != null) {
-            program.setFloat("daylight", worldRenderer.getDaylight(), true);
+        if (worldRenderer != null && backdropProvider != null) {
```

Comment:
Meanwhile, I had a deeper look into the ShaderParameters classes. They contain a lot of logic in them rather than simple values that could be read, say, from a json file if that's what you had in mind. But maybe you have a better idea. I just don't think I can write an issue about it at this stage.